### PR TITLE
Add a missing task

### PR DESCRIPTION
**Pull request by robertpustulka**
See cakephp/bake#362 and cakephp/cakephp#11272

### CODE REVIEW

**Comment by dereuromark** [comment]
Tests are failing though.

**Comment by dereuromark** [comment]
It is also existing, thus not really dead :) right?

**Comment by HavokInspiration** [comment]
Oh my bad, you are right, the task does not exist. But there is a command (I thought you wanted to remove the symfony command).

The right move would be to instead create the task, like there are tasks matching the symfony commands.

**Comment by robertpustulka** [comment]
@ravage84 :smiley:

**Comment by ADmad** [comment]
@ravage84 We made @robertpustulka stop dead in his tracks and make a "U" turn :smile:

**Comment by robertpustulka** [comment]
Why is the plugin implemented in symfony anyway? :smile:

**Comment by HavokInspiration** [comment]
The symfony part is because of phinx which is the tool this plugin is built around.
Since we are using the phinx commands with our shell system, "our" tasks are here to provide the output in case you ask for `--help`.

File: src/Shell/Task/DumpTask.php [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+ * For full copyright and license information, please see the LICENSE.txt
+ * Redistributions of files must retain the above copyright notice.
+ *
+ * @copyright     Copyright (c) Cake Software Foundation, Inc. (http://cakefoundation.org)
+ * @link          http://cakephp.org CakePHP(tm) Project
+ * @license       http://www.opensource.org/licenses/mit-license.php MIT License
+ */
+namespace Migrations\Shell\Task;
+
+/**
```
Comment (saeideng) [inline src/Shell/Task/DumpTask.php]: `use igrations\Shell\Task\CommandTask` ?

Comment (robertpustulka) [reply src/Shell/Task/DumpTask.php]: Its the same namespace, so no need to.

Comment (dereuromark) [reply src/Shell/Task/DumpTask.php]: Actually forbidden even with unused use sniffs :)

**Comment by robertpustulka** [comment]
Can it be merged?

File: src/Shell/Task/DumpTask.php (hunk 1 above)
Comment (saeideng) [reply src/Shell/Task/DumpTask.php]: it is needed for increase readability 
+
avoid of future problems like moving class to place 
I see this problem when working on moving cakephp's HTTP  class

Comment (saeideng) [reply src/Shell/Task/DumpTask.php]: @dereuromark  what says cakephp-codesniffer?

Comment (dereuromark) [reply src/Shell/Task/DumpTask.php]: It doesnt have the advanced sniffs other sniff repos has :) But it is not needed and should be removed. IDEs would also mark it as unneeded.

Comment (robertpustulka) [reply src/Shell/Task/DumpTask.php]: @saeideng I can't recall any class in CakePHP with use statemtents for the classes found in the namespace.

File: src/Shell/Task/DumpTask.php [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+ * This task class is needed in order to provide a correct autocompletion feature
+ * when using the CakePHP migrations shell plugin. It has no effect on the
+ * migrations process.
+ */
+class DumpTask extends CommandTask
+{
+
```
Comment (dereuromark) [inline src/Shell/Task/DumpTask.php]: This newline is technically incorrect.

Comment (robertpustulka) [reply src/Shell/Task/DumpTask.php]: Let's add codesniffer in another PR?

Comment (dereuromark) [reply src/Shell/Task/DumpTask.php]: Yeah!

**Pull request merged**